client/core/bond.go

```diff
@@ -908,9 +906,15 @@ func (c *Core) PostBond(form *PostBondForm) (*PostBondResult, error) {
 			"(target tier %d, bond asset %d, maxBonded %v). "+
 			"Consider using UpdateBondOptions instead.",
 			targetTier, autoBondAsset, wallet.amtString(maxBondedAmt))
-	} else if maintain { // new account with tier maintenance enabled
+	} else if maintain { // new account (or registering a view-only acct) with tier maintenance enabled
+		maxBondedAmt := 4 * form.Bond // default
+		if form.MaxBondedAmt != nil {
+			maxBondedAmt = *form.MaxBondedAmt
+		}
 		dc.acct.authMtx.Lock()
+		dc.acct.bondAsset = bondAssetID
 		dc.acct.targetTier = form.Bond / bondAsset.Amt
+		dc.acct.maxBondedAmt = maxBondedAmt
 		dc.acct.authMtx.Unlock()
 		// maxBondedAmt and bondAsset were set by connectDEX
```

Comment:
no longer correct if view-only acct?

Reply:
No longer correct in all cases. They’re only set here now, if maintain is true and it’s a fresh registration attempt (!acctExists).